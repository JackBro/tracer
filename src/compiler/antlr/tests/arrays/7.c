void main(){
	int a[5], x;

	a[x+1] = a[x] + 2;

	a[x] = a[x+1] + 3;
}

