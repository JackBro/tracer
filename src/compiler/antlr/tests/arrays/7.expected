trans(func_main, i5,
	[p$init_0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X],[SA],[SReturnValue,SX],
	[p$init_1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X],[SA],[SReturnValue,SX],
	[],
	[],
	[],
	not_branch).
trans(func_main, i6,
	[p$init_1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X],[SA],[SReturnValue,SX],
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X],[SA],[SReturnValue,SX],
	[],
	[call(func___TRACER_INIT,[RV0|_],[SRV0|_],[])],
	[],
	not_branch).
trans(func_main, i1,
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X],[SA],[SReturnValue,SX],
	[p4],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A$Next$],[ReturnValue,X],[SA$Next$],[SReturnValue,SX],
	[arrassign(SA$1$,upd(SA,plus(SX,1),plus(ref(SA,SX),2))),arrassign(SA$Next$,upd(SA$1$,SX,plus(ref(SA$1$,plus(SX,1)),3)))],
	[],
	[ref(A,X+1,ARef0),ref(A,X,ARef1),ref(A$1$,X,A$1$Ref0),ref(A$1$,X+1,A$1$Ref1)],
	not_branch) :-
	A$1$=upd(A,X+1,ARef1+2),A$Next$=upd(A$1$,X,A$1$Ref1+3).
trans(func_main, i2,
	[p4],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X],[SA],[SReturnValue,SX],
	[p5],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue$Next$,X],[SA],[SReturnValue$Next$,SX],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
trans(func___TRACER_INIT, i3,
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[p1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[],
	[],
	[],
	not_branch).
trans(func___TRACER_INIT, i4,
	[p1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[p2],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue$Next$],[],[SReturnValue$Next$],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
%--------------------------------------------------
block(b1, func_main, [_], [_], _, nl, [i1,i2,i5,i6],  [n,n],[],[u],[u,n]).
block(b2, func___TRACER_INIT, [_], [_], _, nl, [i3,i4],  [n,n],[],[],[u]).
%--------------------------------------------------
start(func_main, [p$init_0]).
start(func___TRACER_INIT, [p0]).
%--------------------------------------------------
bottom(func_main, [p5]).
bottom(func___TRACER_INIT, [p2]).
%--------------------------------------------------
freshglobalarrays([SH$1$,SH$2$]).
%--------------------------------------------------
freshglobalvars([]).
%--------------------------------------------------
freshlocalarrays(func_main,[SA]).
freshlocalarrays(func___TRACER_INIT,[]).
%--------------------------------------------------
freshlocalvars(func_main,[SReturnValue,SX]).
freshlocalvars(func___TRACER_INIT,[SReturnValue]).
%--------------------------------------------------
freshparams(func_main,[lv(0)]).
freshparams(func___TRACER_INIT,[lv(0)]).
%--------------------------------------------------
init_goal(func_main, [p$init_0], i5, [H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X],[SA],[SReturnValue,SX],[]).
